src/Runner.Worker/Handlers/ScriptHandler.cs

```diff
@@ -144,13 +144,6 @@ public async Task RunAsync(ActionRunStage stage)
             var githubContext = ExecutionContext.ExpressionValues["github"] as GitHubContext;
             ArgUtil.NotNull(githubContext, nameof(githubContext));
 
-            // Add Telemetry to JobContext to send with JobCompleteMessage
-            if (stage == ActionRunStage.Main)
-            {
-                ExecutionContext.StepTelemetry.IsEmbedded = ExecutionContext.IsEmbedded;
```

Comment:
these will get set during `ExecutionContext.CreateChild()` and `Handler.PrepareExecution()`